janus_server/src/bin/aggregation_job_creator.rs

```diff
@@ -0,0 +1,705 @@
+                    return true;
+                }
+                // We don't need to send on the channel: dropping the sender is enough to cause the
+                // receiver future to resolve with a RecvError, which will trigger shutdown.
+                info!(?task_id, "Stopping job creation worker");
+                false
+            });
+
+            // Start job creation tasks for newly-discovered tasks.
+            for (task_id, task) in tasks {
+                if job_creation_task_shutdown_handles.contains_key(&task_id) {
+                    continue;
+                }
+                info!(?task_id, "Starting job creation worker");
+                let (tx, rx) = oneshot::channel();
+                job_creation_task_shutdown_handles.insert(task_id, tx);
+                tokio::task::spawn({
+                    let this = self.clone();
+                    async move { this.run_for_task(rx, task).await }
+                });
+            }
+        }
+    }
+
+    #[tracing::instrument(skip(self))]
+    async fn run_for_task(&self, mut shutdown: Receiver<()>, task: Task) {
+        debug!(task_id = ?task.id, "Job creation worker started");
+        let first_tick_instant = Instant::now()
+            + Duration::from_secs(
+                thread_rng().gen_range(0..self.aggregation_job_creation_interval.as_secs()),
+            );
+        let mut aggregation_job_creation_ticker =
+            time::interval_at(first_tick_instant, self.aggregation_job_creation_interval);
+
+        loop {
+            select! {
+                _ = aggregation_job_creation_ticker.tick() => {
+                    info!(task_id = ?task.id, "Creating aggregation jobs for task");
+                    if let Err(err) = self.create_aggregation_jobs_for_task(&task).await {
+                        error!(task_id = ?task.id, ?err, "Couldn't create aggregation jobs for task")
+                    }
+                }
+
+                _ = &mut shutdown => {
+                    debug!(task_id = ?task.id, "Job creation worker stopped");
+                    return;
+                }
+            }
+        }
+    }
+
+    #[tracing::instrument(skip(self), err)]
+    async fn create_aggregation_jobs_for_task(&self, task: &Task) -> anyhow::Result<()> {
+        match task.vdaf {
+            janus_server::task::Vdaf::Prio3Aes128Count => {
+                self.create_aggregation_jobs_for_task_no_param::<Prio3Aes128Count>(task)
+                    .await
+            }
+
+            janus_server::task::Vdaf::Prio3Aes128Sum { .. } => {
+                self.create_aggregation_jobs_for_task_no_param::<Prio3Aes128Sum>(task)
+                    .await
+            }
+
+            janus_server::task::Vdaf::Prio3Aes128Histogram { .. } => {
+                self.create_aggregation_jobs_for_task_no_param::<Prio3Aes128Histogram>(task)
+                    .await
+            }
+
+            _ => panic!("VDAF {:?} is not yet supported", task.vdaf),
```

Comment:
I think `panic!` here is the right thing to do because this most likely indicates a mismatch between the deployed DB schema and the schema version `aggregation_job_creator` was built against. However I think we should record a message with `error!` so that we get some indication that this is happening besides some pod getting stuck in a crash loop.

Reply:
Added an `error!`.

Note that DB schema skew isn't the only reason; I think we could also encounter this branch if we configured a VDAF that didn't yet have code support (like `poplar1`).